src/neptune/internal/backends/swagger_client_wrapper.py

```diff
@@ -102,11 +105,8 @@ def handle_neptune_http_errors(response, exception: Optional[HTTPError] = None):
         except AttributeError:
             pass
 
-        if exception is None:
-            # raise generic HTTPError with response info
-            raise HTTPError(
-                message=f"{response.status_code} Error: {response.reason} for {response.url}", response=response
-            )
+        # raise generic NeptuneApiException with response info
+        raise NeptuneApiException(f"{response.status_code} Error: {response.reason} for {response.url}")
```

Comment:
Again let's wrap it in `exept AttributeError` and just stringify all arguments if somthing goes wrong.